api/api-internal-agent-v1/src/main/java/com/thoughtworks/go/apiv1/internalagent/InternalAgentControllerV1.java

```diff
@@ -0,0 +1,146 @@
+            post(Routes.InternalAgent.REPORT_COMPLETING, mimeType, this::reportCompleting);
+            post(Routes.InternalAgent.REPORT_COMPLETED, mimeType, this::reportCompleted);
+            post(Routes.InternalAgent.IS_IGNORED, mimeType, this::isIgnored);
+            post(Routes.InternalAgent.GET_COOKIE, mimeType, this::getCookie);
+            post(Routes.InternalAgent.GET_WORK, mimeType, this::getWork);
+        });
+    }
+
+    public String ping(Request request, Response response) {
+        PingRequest pingRequest = PingRequestRepresenter.fromJSON(request.body());
+        ensureAgentIsMakingARequestForItself(pingRequest, request);
+
+        AgentInstruction agentInstruction = buildRepositoryMessageProducer.ping(pingRequest.getAgentRuntimeInfo());
+
+        return AgentInstructionRepresenter.toJSON(agentInstruction);
+    }
+
+    public String reportCurrentStatus(Request request, Response response) {
+        ReportCurrentStatusRequest req = ReportCurrentStatusRequestRepresenter.fromJSON(request.body());
+        ensureAgentIsMakingARequestForItself(req, request);
+
+        buildRepositoryMessageProducer.reportCurrentStatus(req.getAgentRuntimeInfo(), req.getJobIdentifier(),
+                req.getJobState());
+
+        return NOTHING;
+    }
+
+    public String reportCompleting(Request request, Response response) {
+        ReportCompleteStatusRequest req = ReportCompleteStatusRequestRepresenter.fromJSON(request.body());
+        ensureAgentIsMakingARequestForItself(req, request);
+
+        buildRepositoryMessageProducer.reportCompleting(req.getAgentRuntimeInfo(), req.getJobIdentifier(),
+                req.getJobResult());
+
+        return NOTHING;
+    }
+
+    public String reportCompleted(Request request, Response response) {
+        ReportCompleteStatusRequest req = ReportCompleteStatusRequestRepresenter.fromJSON(request.body());
+        ensureAgentIsMakingARequestForItself(req, request);
+
+        buildRepositoryMessageProducer.reportCompleted(req.getAgentRuntimeInfo(), req.getJobIdentifier(),
+                req.getJobResult());
+
+        return NOTHING;
+    }
+
+    public String isIgnored(Request request, Response response) {
+        IsIgnoredRequest isIgnoredRequest = IsIgnoredRequestRepresenter.fromJSON(request.body());
+        ensureAgentIsMakingARequestForItself(isIgnoredRequest, request);
+
+        boolean isIgnored = buildRepositoryMessageProducer.isIgnored(isIgnoredRequest.getAgentRuntimeInfo(), isIgnoredRequest.getJobIdentifier());
+
+        return valueOf(isIgnored);
+    }
+
+    public String getCookie(Request request, Response response) {
+        GetCookieRequest getCookieRequest = GetCookieRequestRepresenter.fromJSON(request.body());
+        ensureAgentIsMakingARequestForItself(getCookieRequest, request);
+
+        return buildRepositoryMessageProducer.getCookie(getCookieRequest.getAgentRuntimeInfo());
```

Comment:
@maheshp If I’m not mistaken, this returns raw text? For consistency, do you think we should quote it to be valid JSON?

Reply:
You are right, I will change it.